packages/drivers/routerlicious-driver/src/queryStringUtils.ts

```diff
@@ -0,0 +1,21 @@
+export type QueryStringType = Record<string, string | number | boolean>;
+
+/**
+ * Generates URL string from the given URL and query parameters.
+ * @param url - URL to which query params should be appended. Can include base/default query params.
+ * @param queryParams - Query parameters from which to create a query. Will override any query params in url.
+ */
+export function buildUrlWithQueryString(
+	url: URL | string,
+	queryParams: QueryStringType,
+): string {
+	// Initialize urlSearchParams with query params from the base URL itself
+	const outputUrl = new URL(url);
```

Comment:
Good question, and I actually ran into this issue when running the tests locally. No change would be required here, since as soon as there is an invalid input, Line 13 above throws with appropriate error message and eventually fails the complete call stack.